### PR TITLE
feat: update vehicle overlay plugin

### PR DESCRIPTION
**Pull request by KhalilSelyan**
## Description

This PR introduces a series of enhancements to the `awf_vehicle_rviz_plugin` for Autoware, aimed at improving the visual elements and functionality of the plugin. The modifications include adjustments to opacity, color specifications for signal indicators, changes to the gear indicator outline, and improvements to the speed limit and traffic light indicators. These changes are designed to make the plugin more intuitive, visually appealing, and user-friendly, enhancing the driving and simulation experience in Autoware.

## Related links

- Modernize the RViz2 overlay interface [#3984](https://github.com/autowarefoundation/autoware/issues/3984)
- feat: update rviz splash and vehicle UI display [autoware_launch#836 (comment)](https://github.com/autowarefoundation/autoware_launch/pull/836#issuecomment-1917530493)
- Improve the new awf_vehicle_rviz_plugin [#4146](https://github.com/autowarefoundation/autoware/issues/4146)
- Reference Figma page link: [Figma Design](https://www.figma.com/file/xjrQXpUX0LbRNa14zKiy98/autoware-hud?type=design&node-id=17%3A263&mode=design&t=W5r8RWRr2fEbZ0W6-1)

## Notes for reviewers

This PR is a follow-up from discussions in issues #3984 and comments in PR #836 as well as #4146, focusing on enhancing the RViz plugin's aesthetics and usability. Reviewers are encouraged to check the visual elements and functionality changes against the proposed design specifications.

## Interface changes

- Renamed the package from `awf_vehicle_rviz_plugin` to `autoware_vehicle_overlay_rviz_plugin`.
- Renamed `rviz_2d_overlay_msgs` to `autoware_overlay_msgs`.
- Updated the top-level folder name to `autoware_overlay_rviz_plugin`.

## Effects on system behavior

The changes in this PR will not affect the core functionalities of Autoware but will improve the user interaction with the RViz visualization plugin by providing a more intuitive and visually appealing interface.

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

## In-review checklist for the PR reviewers

- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

After all checkboxes are checked and discussions are resolved, the PR author should proceed with the merging process.

- [x] There are no open discussions, or they are tracked via tickets.
- [x] The PR is ready for merge.

### CODE REVIEW

**Comment by KhalilSelyan** [comment]
https://github.com/autowarefoundation/autoware.universe/assets/36904941/7c1a492e-db7a-401e-a9de-27bd6b9002b2

**Comment by xmfcx** [comment]
Thanks for all the updates, it really looks much more polished now!

I've updated the package name to "autoware_overlay_rviz_plugin" from the issue.

**Comment by KhalilSelyan** [comment]
Thanks for helping on the design, it was fun to work on

File: common/autoware_overlay_rviz_plugin/autoware_overlay_rviz_plugin/src/signal_display.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ SignalDisplay::SignalDisplay()
   property_top_ = new rviz_common::properties::IntProperty(
     "Top", 10, "Top position of the overlay", this, SLOT(updateOverlayPosition()));
   property_signal_color_ = new rviz_common::properties::ColorProperty(
-    "Signal Color", QColor(94, 130, 255), "Color of the signal arrows", this,
+    "Signal Color", QColor("25, 18, 89"), "Color of the signal arrows", this,
```
Comment (xmfcx) [inline common/autoware_overlay_rviz_plugin/autoware_overlay_rviz_plugin/src/signal_display.cpp:50]:
```suggestion
    "Signal Color", QColor(QString("#00E678")), "Color of the signal arrows", this,
```
Screenshot from 2024-02-06 11-37-18
 
Current color is darker than the proposed `#3F2EE3` from the [Autoware website color palette](https://coolors.co/3f2ee3-00c8ff-00e678-ff5c0f-00002a-0094ff). But even when I updated it to that color, it still lacked the contrast.

https://coolors.co/contrast-checker/3f2ee3-000000

This color doesn't work on black backgrounds :(

Instead let's use `#00E678` from [the palette](https://coolors.co/3f2ee3-00c8ff-00e678-ff5c0f-00002a-0094ff) then.

https://coolors.co/contrast-checker/00E678-000000

It has much better contrast.

Also you can use hex strings directly in QColor() constructor.

**Comment by xmfcx** [comment]
When the speed is less than `0.7 * v_max`, it should be the `color_min` (`#FF9999`), not light gray.

Screenshot from 2024-02-06 11-57-50

Also when the velocity is positive but the speed limit is 0, the circle disappears.
Screenshot from 2024-02-06 11-58-31

**Comment by KhalilSelyan** [comment]
> When the speed is less than `0.7 * v_max`, it should be the `color_min` (`#FF9999`), not light gray.
> 
> Screenshot from 2024-02-06 11-57-50
> 
> Also when the velocity is positive but the speed limit is 0, the circle disappears. Screenshot from 2024-02-06 11-58-31

I don't see the circle disappearing in my setup, can you check if you're the latest commit ?

Also, maybe i misunderstood the logic currently it is this way 
- 0 to 70% -> light gray
- 70% to 100% -> light red interpolating slowly towards red

should it be
0-70% -> light red
70-100% -> light red interpolating slowly towards red

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/pull/6323/commits/75166f93443f10cb6e803fdeddda2904f63064be

Here I've updated the traffic light colors.

https://github.com/autowarefoundation/autoware.universe/assets/10751153/dda977b7-6bba-4d5a-ad79-fa9368f38425

Set it to no pen to disable strokes.

Set the colors to be more consistent (uniform Saturation and Values).
Set the scale of the small traffic light image to be 1.0. (And color to be darker)

This way when it is not used, it is a more muted color.

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/assets/10751153/69f35423-24b0-45b0-a4c9-95ae6b9e4425

This is the final state after all the updates, I think we can merge it!

**Comment by KhalilSelyan** [comment]
https://github.com/autowarefoundation/autoware_launch/pull/855 i created the PR for the autoware.rviz file as well

**Comment by xmfcx** [comment]
Also could you update the README with speed limit and traffic light information too?

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/pull/6323/commits/5caa6a7e36b68267b6c86dc9097d0e5ebeafb726

With this commit, I've converted the color space transition to be in HSV.

Initial plan was:
>   - Speed limit logo outer border to fade from `#FF9999` to `#FF3333` depending on:
>    - color_min = `#FF9999` = `hsl(0, 100%, 80%)`
>    - color_max = `#FF3333` = `hsl(0, 100%, 60%)`
>    - `v : [0, v_max * 0.7) -> color_min`
>    - `v : [v_max * 0.7, v_max] -> interpolate between color_min and color_max (lightness of HSL)`

It turns QT has HSV and not HSL. Converting them to HSV:
- color_min = `#FF9999` = `hsl(0, 100%, 80%)` = `hsv(0, 40%, 100%)`
- color_max = `#FF3333` = `hsl(0, 100%, 60%)` = `hsv(0, 80%, 100%)`

Also I've lowered the thresholds by 0.1, previous limits were:
- `speed_to_limit_ratio_min = 0.7`
- `speed_to_limit_ratio_max = 1.0`
new limits:
- `speed_to_limit_ratio_min = 0.6`
- `speed_to_limit_ratio_max = 0.9`

With these, the interpolation works similar to before but it is easier to tune now.

**Review by xmfcx — APPROVED** [review]
Thanks for everything!

**Pull request merged**